packages/query-core/src/query.ts

```diff
@@ -499,7 +499,11 @@ export class Query<
           return
         }
 
-        this.setData(data)
+        try {
+          this.setData(data)
+        } catch (error) {
+          onError(error as TError)
+        }
```

Comment:
note: we're missing a `return` here; otherwise, we'll go on with calling the `onSuccess` callback, which isn't good if we're in error state.

Reply:
sure, my bad